core/lib/zksync_core/src/state_keeper/batch_executor/mod.rs

```diff
@@ -445,58 +445,30 @@ impl BatchExecutor {
         Vec<CompressedBytecodeInfo>,
         Vec<Call>,
     ) {
-        // Note, that the space where we can put the calldata for compressing transactions
-        // is limited and the transactions do not pay for taking it.
-        // In order to not let the accounts spam the space of compressed bytecodes with bytecodes
-        // that will not be published (e.g. due to out of gas), we use the following scheme:
-        // We try to execute the transaction with compressed bytecodes.
-        // If it fails and the compressed bytecodes have not been published,
-        // it means that there is no sense in polluting the space of compressed bytecodes,
-        // and so we re-execute the transaction, but without compression.
-
-        // Saving the snapshot before executing
-        vm.make_snapshot();
```

Comment:
It will lead to discrepancy in EN execution

Reply:
@Deniallugo it won't, because if tx has failed to publish compressed bytecode, it won't even make it to the external ndoe

Reply:
If we are talking about executing only future txs, then yes. 
but for the past txs it's breaking

Reply:
Changed code, so EN behaviour is the same as it was before